editor/src/clj/editor/properties_view.clj

```diff
@@ -97,18 +97,33 @@
     (ui/auto-commit! update-fn)
     (select-all-on-click!)))
 
+(defn set-text-field-icon!
+  [text-field image-url]
+  (let [style (str "-fx-background-image: url('" image-url "');"
+                   "-fx-background-repeat: no-repeat;"
+                   "-fx-background-position: right 6px center;")]
+    (.setStyle text-field style)))
```

Comment:
I spent a few hours fighting with javafx and trying to set everything using functions instead of styles, but it seems like we just override it with styles later. Because no matter what I tried I didn't see icons. If you have better ideas how to do it, let me know because I'm not an expert in JavaFX .